packages/hidp/tests/unit_tests/test_api/test_passwords.py

```diff
@@ -0,0 +1,263 @@
+        # unchanged
+        client2 = APIClient(enforce_csrf_checks=True)
+        client2.force_login(self.verified_user)
+
+        new_password = "NewP@ssw0rd!"
+
+        response = self.client.post(
+            self.url,
+            data={
+                "token": default_token_generator.make_token(self.verified_user),
+                "new_password": new_password,
+            },
+        )
+
+        self.verified_user.refresh_from_db()
+        self.assertTrue(self.verified_user.check_password(new_password))
+
+        # Session that initiated the password change no longer has the old session hash
+        self.assertNotEqual(
+            self.client.session.get("_auth_user_hash"), pre_password_change_session_hash
+        )
+        # The other session still has the old session hash and is thus no longer valid
+        self.assertEqual(
+            client2.session.get("_auth_user_hash"), pre_password_change_session_hash
+        )
+
+        # The session that initiated the password change has the updated session hash
+        self.assertEqual(
+            self.client.session.get("_auth_user_hash"),
+            self.verified_user.get_session_auth_hash(),
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.NO_CONTENT)
+        self.assertIsNone(response.data)
+
+    def test_password_reset_confirmation_invalid_token(self):
+        """
+        Verify behavior when an invalid token is provided.
+
+        - The user's password remains unchanged
+        - Session hash remains unchanged
+        - The response status code is 400 Bad Request
+        - The response contains the appropriate error message
+        """
+        self.client.force_login(self.verified_user)
+        pre_password_change_session_hash = self.verified_user.get_session_auth_hash()
+
+        response = self.client.post(
+            self.url,
+            data={
+                "token": "invalid-token",
+                "new_password": "NewP@ssw0rd!",
+            },
+        )
+
+        self.verified_user.refresh_from_db()
+        self.assertTrue(self.verified_user.check_password("P@ssw0rd!"))
+        self.assertEqual(
+            self.verified_user.get_session_auth_hash(), pre_password_change_session_hash
+        )
+
+        self.assertEqual(response.status_code, HTTPStatus.BAD_REQUEST)
+        errors = response.json()["token"]
+        self.assertEqual(len(errors), 1)
+        self.assertEqual(str(errors[0]), "Invalid or expired token.")
+
+    @override_settings(
+        AUTH_PASSWORD_VALIDATORS=[
+            {
+                "NAME": "django.contrib.auth.password_validation.MinimumLengthValidator",  # noqa: E501
+                "OPTIONS": {
+                    "min_length": 10,
+                },
+            }
+        ]
+    )
+    def test_password_reset_confirmation_invalid_password(self):
+        """
+        Verify behavior when an invalid token is provided.
```

Comment:
Surely this should be 
```suggestion
        Verify behavior when an invalid password is provided.
```